### ISSUE TITLE
Should incomplete imports be instantiated?

### ISSUE THREAD

**Issue by kerimoyle** [cellml/libcellml#663]
We have a test in the `Parser` (see below).  When #653 is merged it will be possible to create imported entities for two units and two components instead of the one each that we currently instantiate.  They would have missing `importReference` attributes, but the objects would be constructed.  Is this something we should be doing, since we want to parse as much as possible from even invalid models?

```
TEST(Parser, invalidImportsAndGetIssue)
{
    const std::string in =
        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
        "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"some-other-model.xml\" sauce=\"hollandaise\">\n"
        "    <units units_ref=\"a_units_in_that_model\" name=\"units_in_this_model\"/>\n"
        "    <component component_ref=\"a_component_in_that_model\" name=\"component_in_this_model\"/>\n"
        "    <invalid_nonsense/>\n"
        "    <units units_ruff=\"dog\" name=\"fido\"/>\n"
        "    <component component_meow=\"cat\" name=\"frank\"/>\n"
        "  </import>\n"
        "</model>\n";

... etc ... 

    const std::string output =
        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
        "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"some-other-model.xml\">\n"
        "    <component component_ref=\"a_component_in_that_model\" name=\"component_in_this_model\"/>\n"
        "    <units units_ref=\"a_units_in_that_model\" name=\"units_in_this_model\"/>\n"
        "  </import>\n"
        "</model>\n";
```

This is the result of only setting an import source when there's a valid `component_ref` or `units_ref`:
```
while (childAttribute) {
                if (childAttribute->isType("name")) {
                    importedComponent->setName(childAttribute->value());
                } else if (childAttribute->isType("id")) {
                    importedComponent->setId(childAttribute->value());
                } else if (childAttribute->isType("component_ref")) {
                    importedComponent->setSourceComponent(importSource, childAttribute->value());   // <<< bug comes from here
                } else {
                    IssuePtr issue = Issue::create();
                    issue->setDescription("Import of component '" + childNode->attribute("name") + "' from '" + node->attribute("href") + "' has an invalid attribute '" + childAttribute->name() + "'.");
                    issue->setImportSource(importSource);
                    mParser->addIssue(issue);
                    errorOccurred = true;
                }
```

**Comment by nickerso** [comment]
yes, I think that as with other objects we should try to construct as much as possible when parsing.

**Comment by kerimoyle** [comment]
Closed via #667
